### ISSUE TITLE
Improve pc_align output rotation matrix

### ISSUE THREAD

**Issue by oleg-alexandrov** [NeoGeographyToolkit/StereoPipeline#191]
pc_align outputs the rotation + translation matrix that brings the source points on top of the reference points. This is in respect to the center of the planet, and is not the intuitive translation and rotation we have in mind relative to a point on the planet surface. 

The problem is that the planet center is very far. A rotation around that center is basically a translation on the surface of the planet. Hence, if say the rotation is by an angle alpha, one could equally successfully rotate by 1.5*alpha, then do a translation back to compensate for it. Hence, both the rotation and translation component of the transform we output, while correct, is a little meaningless. 

We already output an "intuitive" translation vector. For that, we find the position of the center of gravity of the source points before the alignment transform, then after the transform, and subtract the second from the first.  To get the intuitive local rotation, one would take the source cloud, shift it by that translation vector we compute to bring it on top of the transformed source cloud, and now find the local rotation between these two clouds which now have the same center of gravity. Then you would convert that one to NED at that center of gravity.

**Comment by oleg-alexandrov** [comment]
Also, the current Euler angles are in respect to the ECEF, and should be in respect to the "local" rotation.

**Comment by oleg-alexandrov** [comment]
After a study, I realized the obvious. While we can compute a "local" translation applied by pc_align, which is the vector from the centroid of the source points before the alignment transform to the centroid of the same points after the alignment transform, there is no way to express the rotation component of the pc_align transform in a "local" coordinate system. The rotation component of a rigid transform is still the same, whether expressed around one rotation center (planet center) or a different one (local coordinate system). This comes from the obvious math:

R*x + t  = R*(x-x0) + (R*x0 + t)

hence if one rotates around a new point the R is same, but the translation changes. 

I fixed a bug along the way, the translation vector was wrong when expressed in North-East-Down coordinates. Now it is correct.

David, if you still feel it is relevant to express the rotation matrix in the NED coordinate system in addition to ECEF, do let me know. This looks easy enough now: 

NED_Rotation = (NED_to_ECEF_Rotation)^-1 * ECEF_Rotation * NED_to_ECEF_rotation. 

I just want to know if this is still useful in light of the above discussion.  The vast majority of the time all rotation angles in any of these coordinate systems will be quite small, like what you observed, 0.02 degrees or whatever that number was since the alignment transform between any two realistic DEMs on the planet surface is mostly a translation anyway.

**Comment by dpmayerUSGS** [comment]
@oleg-alexandrov Yes, I would definitely find it useful to also see the rotations expressed in the NED coordinate system, even if the absolute angles are small. It would be more intuitive for cases where I need to understand the manner in which a source cloud moved to come into alignment with a reference.

**Comment by oleg-alexandrov** [comment]
This was easy enough to implement, and will show up in the daily build at 

https://byss.arc.nasa.gov/stereopipeline/daily_build/

when the date is 2018-08-22 or later, which should be hopefully tomorrow. 

The trickier part is the testing. I took a DEM, and applied to it a rotation around the z axis of the planet of 2 degrees, then computed the Euler angles in degrees of the ECEF transform, and the Euler angles in the NED coordinate system. This is what pc_align is outputting:

Euler angles (degrees): Vector3(0,0,2)
Euler angles (North-East-Down, degrees): Vector3(0.713814,0.0116385,-1.86818)

The DEM was at a latitude of about 69 degrees north. This is quite North, so the Down direction on the planet center is more aligned to the z direction. So it makes sense that the NED Euler angle around the Down axis (third component is largest). 

Then I made a cloud on the equator.  Here, the North direction is the same as the z direction. And what I got as results was:

Euler angles (degrees): Vector3(0,0,2)
Euler angles (North-East-Down, degrees): Vector3(2,1.09635e-05,-0.000628097)

which again makes sense. 

David, I assume you would want to do your own finer grained testing for a DEM for which you have a feel for the result.
